Firestore/Source/Local/FSTPersistence.h

```diff
@@ -99,6 +100,31 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (void)commitGroup:(FSTWriteGroup *)group;
 
+@property (nonatomic, readonly, assign) const FSTTransactionRunner& run;
+
 @end
 
+@protocol FSTTransactional
+
+- (void)startTransaction;
+
+- (void)commitTransaction;
+
+@end
+
+struct FSTTransactionRunner {
+  template <typename F>
+  auto operator() (F block) const -> decltype(block()) {
```

Comment:
I am also getting a lot of warnings about missing nullability specifiers on this line. I'm not sure what the right thing to add here is though. `_Nonnull F` ? I think there might a similar issue with the return type as well...